src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { memo, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoPatch } from '../../types/TodoPatch';
+
+type Props = {
+  todo: Todo,
+  onDelete: (todoId: number) => void,
+  onUpdateTodo?: (todoId: number, data: TodoPatch) => void,
+  isLoading: boolean
+};
+
+export const TodoItem: React.FC<Props> = memo(
+  ({
+    todo,
+    onDelete,
+    onUpdateTodo = () => { },
+    isLoading,
+  }) => {
+    const {
+      id,
+      title,
+      completed,
+    } = todo;
+
+    const [isEditing, setIsEditing] = useState(false);
+    const [newTitle, setNewTitle] = useState(title);
+
+    const handleDeleteTodo = () => onDelete(id);
+
+    const handleToggleTodo = () => onUpdateTodo(id, { completed: !completed });
```

Comment:
If your `onUpdateTodo` is not a mandatory `prop` (so it is written in your type `Props`), then it is worth checking first whether `onUpdateTodo` has now been passed, and only then use it.